### DISCUSSION TITLE
2D Skeleton Implentation in 3D Spaces

### DISCUSSION THREAD

**Discussion by BunkWire2X8** [godotengine/godot-proposals#8077] [3D]
For 2.5D games or 3D ones that want to implement it for other use cases, I think that allowing 3D support for 2D skeletons would be incredibly helpful.

While I know that something like this could _probably_ already be rudimentarily implemented with SubViewports, I'd imagine that way is probably not the most optimized. This could become more of a problem depending on how many 2D skeletons one would like to add to their scenes. Having to stick to this way could also restrict certain use cases and make certain things just generally more difficult to implement.

A [similar suggestion](https://github.com/EsotericSoftware/spine-runtimes/issues/2135) was also proposed to the [spine-godot](https://github.com/EsotericSoftware/spine-runtimes/tree/4.1/spine-godot) runtime to support Spine's 2D skeletal system in 3D as the most voted Godot suggestion, but the idea seems to be currently on the backburner for now as they continue their work on the other Spine runtimes. Spine implementation or not though, I think that having something like this be implemented in base Godot would generally be more convenient.

**Comment by ItzASecretBoi** [comment]
🙏🏽  please I need this,  my kidneys need this too

**Comment by ItzASecretBoi** [comment]
By the way if anyone interested in this, someone has done something INSANELY simular in godot but it's paid. 

https://bunkwire2x8.itch.io/paperskeleton

**Reply by Calinou** [reply]
> someone has done something INSANELY simular in godot but it's paid.

That someone is this proposal's author :slightly_smiling_face:

**Reply by ItzASecretBoi** [reply]
😲 this is the biggest plot twist of all time.
